lib/reporting/reports/sales_tax/sales_tax_totals_by_producer.rb

```diff
@@ -0,0 +1,167 @@
+# frozen_string_literal: true
+
+# rubocop:disable Metrics/ClassLength
+module Reporting
+  module Reports
+    module SalesTax
+      class SalesTaxTotalsByProducer < Base
+        def search
+          report_line_items.orders
+        end
+
+        def order_permissions
+          return @order_permissions unless @order_permissions.nil?
+
+          @order_permissions = ::Permissions::Order.new(user, ransack_params)
+        end
+
+        def report_line_items
+          @report_line_items ||= Reporting::LineItems.new(order_permissions, params)
+        end
+
+        def query_result
+          # The objective is to group the line items by
+          # [tax_rate, supplier_id, distributor_id and order_cycle_id]
+          report_line_items.list
+            .left_joins(product: { tax_category: :tax_rates })
```

Comment:
For some reason, this join generates distinct line_items. If a line item matches 2 tax_rates, it'll be shown only one time.
I tried a similar query on Spree::LineItem, and I got the correct results.
```ruby
Spree::LineItem.where('CONDITION')
            .left_joins(product: { tax_category: :tax_rates })
            .select('spree_tax_rates.id as spree_tax_rate_id', 'spree_line_items.*')
            .group_by do |line_item|
            [
              line_item.spree_tax_rate_id,
              line_item.supplier_id,
              line_item.order.distributor_id,
              line_item.order.order_cycle_id
            ]
```
I'll need to inspect `Reporting::LineItems` to revert that.